File: FlinkTurbineFaultDiagnosis/src/main/java/com/wang/flink/cleandata/BatchWordCount.java
package com.wang.flink.cleandata;

import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.AggregateOperator;
import org.apache.flink.api.java.operators.DataSource;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.util.Collector;


/**
 * @author 王一宁
 * @date 2020/1/1 21:00
 * 批处理
 */
public class BatchWordCount {
    public static void main(String[] args) throws Exception{
        //获取配置
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();

        //获取数据源
        String inputPath = "F:\\input01";
        String outPath = "F:\\output01\\result";

        DataSource<String> text = env.readTextFile(inputPath);
        AggregateOperator<Tuple2<String, Integer>> counts = text.flatMap(new Tokenizer()).groupBy(0).sum(1);


        //保存文件
        counts.writeAsCsv(outPath,"\n"," ").setParallelism(1);

        //执行
        env.execute("batch wordcount");
    }

    public static class Tokenizer implements FlatMapFunction<String, Tuple2<String,Integer>>{
        @Override
        public void flatMap(String value, Collector<Tuple2<String, Integer>> out) throws Exception {
            String[] tokens = value.toLowerCase().split("\\W+");
            for (String token : tokens) {
                if (token.length()>0){
                    out.collect(new Tuple2<String,Integer>(token,1));
                }
            }

        }
    }
}

File: FlinkTurbineFaultDiagnosis/src/main/scala/com/wang/turbine/cleandata/BatchWordCount.scala
package com.wang.turbine.cleandata

import org.apache.flink.api.scala.ExecutionEnvironment

/**
 * //取出发电机的时间
 * //发电机各项参数：故障时间、风速、发电机转速、变流器无功功率、控制柜温度、机舱变频柜温度
 * //环境温度、机舱温度、变桨电机123温度、变桨柜123温度、叶片123速度
 * //网侧有功功率、变流器无功功率、理论有功功率
 */
object BatchWordCount {
  def main(args: Array[String]): Unit = {

    val inputPath = "D:\\APP\\IDEA\\workplace\\FlinkTurbineFaultDiagnosis\\src\\main\\resources\\turbine"
    val outPath = "D:\\APP\\IDEA\\workplace\\FlinkTurbineFaultDiagnosis\\src\\log\\result"

    //获取执行环境
    val env = ExecutionEnvironment.getExecutionEnvironment

    val text = env.readTextFile(inputPath)

    //隐士转换
    import org.apache.flink.api.scala._

    val filtered = text.map(line =>{
      val fields = line.split("\t")
      val gw = fields(0)
      val status = fields(2) //数据的可用状态
      val date01 =  fields(1) //发电机的实时时间
      val test001 = fields(8)
      (gw,status,date01,test001)

    })

    filtered.print()

    env.execute()

  }
}
